Cargo.toml

```diff
@@ -80,6 +80,7 @@ sui-gateway = { version = "^1.0.0", path = "packages/sui-gateway" }
 sui-types = { version = "^1.0.0", path = "packages/sui-types" }
 starknet-checked-felt = { version = "^1.0.0", path = "packages/starknet-checked-felt" }
 starknet-types-core = { version = "0.1.7" }
+starknet-types = { version = "^1.0.0", path = "packages/starknet-types" }
```

Comment:
The types in this crate can be substituted, if we use the [cainome](https://github.com/cartridge-gg/cainome) crate. It has an `abigen!()` macro similar to the ether's one, which can auto-generate those types. 
I wasn't sure if you'd like another dependency or have custom implementations.